test/mocha/10-documents.js

```diff
@@ -30,240 +36,308 @@ const key2 = new X25519KeyAgreementKey2020({
+      let result1;
+      let error;
+      try {
+        result1 = await documents._hmacString({
+          hmac,
+          value: '294c9caa-707a-4758-ae5c-fe7306c25cc2'
+        });
+      } catch(e) {
+        error = e;
+      }
+      assertNoError(error);
 
-  it('should produce the same output given the same value', async () => {
-    let result1;
-    let error;
-    const value = '294c9caa-707a-4758-ae5c-fe7306c25cc2';
-    try {
-      result1 = await documents._hmacString({hmac, value});
-    } catch(e) {
-      error = e;
-    }
-    assertNoError(error);
-
-    let result2;
-    error = undefined;
-    try {
-      result2 = await documents._hmacString({hmac, value});
-    } catch(e) {
-      error = e;
-    }
-    assertNoError(error);
-
-    result1.should.eql(result2);
+      let result2;
+      error = undefined;
+      try {
+        result2 = await documents._hmacString({
+          hmac,
+          value: '0e26c923-84e6-4918-9337-f82c56951007'
+        });
+      } catch(e) {
+        error = e;
+      }
+      assertNoError(error);
+
+      result1.should.not.eql(result2);
+    });
   });
+});
 
-  it('should produce different output given different values', async () => {
-    let result1;
-    let error;
-    try {
-      result1 = await documents._hmacString({
-        hmac,
-        value: '294c9caa-707a-4758-ae5c-fe7306c25cc2'
+describe('Documents Database Tests', function() {
+  describe('Indexes', function() {
+    beforeEach(async () => {
+      const collectionName = 'tokenization-registration';
+      await cleanDB({collectionName});
+    });
+    it(`is NOT indexed for 'registration.internalId' in getRegistration()`,
+      async function() {
+      // Note: an index is not created for `registration.internalId` because
+      // queries based on that field alone are rare; they are not run in hot
+      // code paths nor do they need to run quickly -- so the index cost is not
+      // justified. Therefore, a collection scan should be done.
+        const collectionName = 'tokenization-registration';
+        await insertRecord({record: mockDocument, collectionName});
+
+        const {internalId} = mockDocument.registration;
+        const {executionStats} = await documents.getRegistration({
+          internalId, explain: true
+        });
+        executionStats.nReturned.should.equal(1);
+        executionStats.totalKeysExamined.should.equal(0);
+        executionStats.executionStages.inputStage.stage.should
+          .equal('COLLSCAN');
       });
-    } catch(e) {
-      error = e;
-    }
-    assertNoError(error);
-
-    let result2;
-    error = undefined;
-    try {
-      result2 = await documents._hmacString({
-        hmac,
-        value: '0e26c923-84e6-4918-9337-f82c56951007'
+    it(`is properly indexed for compound query of ` +
+    `'registration.externalIdHash' and 'registration.documentHash' in ` +
+    `_getRegistrationRecord()`, async function() {
```

Comment:
Here and elsewhere needs some indentation and quote fixing:

```suggestion
    it('is properly indexed for compound query of ' +
      `'registration.externalIdHash' and 'registration.documentHash' in ` +
      '_getRegistrationRecord()', async function() {
```